### PR TITLE
installation instructions: update

### LINKED ISSUE #170: Install, step 4 issues
Following these instructions: https://actinia.mundialis.de/tutorial/installation.html

Step 4:
```
    mkdir -p $HOME/actinia/workspace/tmp

    cd /tmp

    git clone https://github.com/mundialis/actinia_core.git actinia_core
    virtualenv -p python3.5 actinia_venv
    source actinia_venv/bin/activate
    cd actinia_core
    pip install -r requirements.txt
    python setup.py install
    deactivate
```
---
If I see right I think it's accidentally directing me to root's /tmp instead of the newly created tmp directory? Instead of 
`cd /tmp` I will follow directions using `cd $HOME/actinia/workspace/tmp`

---

This line: 
`virtualenv -p python3.5 actinia_venv` Did not work for me, so far I only have python3.8 installed on my computer. I'm guessing it's related to that. I decided to let it decide which version to use and it defaulted to 3.8 and I will attempt to continue the install like that:`virtualenv -p python3 actinia_venv` 

I needed to also install this library (in order to get uwsgi to install)
   - python3-dev
---

### LINKED ISSUE #171: Install, step 5 issues
I'm following the instructions here: https://actinia.mundialis.de/tutorial/installation.html

```
    cd /tmp

    virtualenv -p python2.7 grass_venv
    source grass_venv/bin/activate
    cd actinia_core
    pip install -r requirements-grass-gis.txt
```

As with step 4, I assume that we are meant to `cd $HOME/actinia/workspace/tmp` and not just /tmp. 
The instructions say we need to have a python3 env for grass ,but the code sample shows 2.7. That should be 3?

There is no file "requirements-grass-gis.txt" in the actinia_core folder. I have one called "requirements.txt" and requirements_ubuntu19.txt".

Both seem to have notes about different versions, possible changes, etc. I will continue with "requirements_ubuntu19.txt" (I'm running Mint 20) and see how it goes.

Went ok except for this error:
`ERROR: sphinx 3.5.4 has requirement docutils<0.17,>=0.12, but you'll have docutils 0.17 which is incompatible.`
Since that's just for documentation, I'll ignore it for now.

### LINKED ISSUE #172: Install, step 6: Cant find files in /etc/default/actinia
I'm following the install instructions here : https://actinia.mundialis.de/tutorial/installation.html 

My OS is Linux Mint 20.1, v4.8.6 . The computer has not been used for much before so it is empty of most additional libraries not explicitly mentioned in the installation instructions.

Step 6 says:

> Make sure that a redis service is running

I don't know what that is so I'll look it up. I would recommend turning this into a link to instructions for starting a redis service. I am now following these instructions: https://redis.io/topics/quickstart . Fairly easy to follow so that worked, and I turned it on.
From here, the instructions state:
```
cd /tmp
source actinia_venv/bin/activate

# Create the superuser
actinia-user create -u superadmin -w abcdefgh -r superadmin -g admin -c 100000000000 -n 1000 -t 6000

# Start the actinia service with
actinia_server --host 0.0.0.0 --port 5000
```

When I try to run actinia-user create, I get an error
```
Traceback (most recent call last):
  File "/home/halota/actinia/workspace/tmp/actinia_venv/bin/actinia-user", line 4, in <module>
    __import__('pkg_resources').run_script('actinia-core==0.99.26.post0.dev1+ge1510fc', 'actinia-user')
  File "/home/halota/actinia/workspace/tmp/actinia_venv/lib/python3.8/site-packages/pkg_resources/__init__.py", line 666, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/home/halota/actinia/workspace/tmp/actinia_venv/lib/python3.8/site-packages/pkg_resources/__init__.py", line 1462, in run_script
    exec(code, namespace, namespace)
  File "/home/halota/actinia/workspace/tmp/actinia_venv/lib/python3.8/site-packages/actinia_core-0.99.26.post0.dev1+ge1510fc-py3.8.egg/EGG-INFO/scripts/actinia-user", line 30, in <module>
    from actinia_core.resources.common.user import ActiniaUser
  File "/home/halota/actinia/workspace/tmp/actinia_venv/lib/python3.8/site-packages/actinia_core-0.99.26.post0.dev1+ge1510fc-py3.8.egg/actinia_core/resources/common/user.py", line 35, in <module>
    from actinia_core.resources.common.redis_user import redis_user_interface
  File "/home/halota/actinia/workspace/tmp/actinia_venv/lib/python3.8/site-packages/actinia_core-0.99.26.post0.dev1+ge1510fc-py3.8.egg/actinia_core/resources/common/redis_user.py", line 28, in <module>
    from .redis_base import RedisBaseInterface
  File "/home/halota/actinia/workspace/tmp/actinia_venv/lib/python3.8/site-packages/actinia_core-0.99.26.post0.dev1+ge1510fc-py3.8.egg/actinia_core/resources/common/redis_base.py", line 29, in <module>
    from .logging_interface import log
  File "/home/halota/actinia/workspace/tmp/actinia_venv/lib/python3.8/site-packages/actinia_core-0.99.26.post0.dev1+ge1510fc-py3.8.egg/actinia_core/resources/common/logging_interface.py", line 40, in <module>
    global_config.read()
  File "/home/halota/actinia/workspace/tmp/actinia_venv/lib/python3.8/site-packages/actinia_core-0.99.26.post0.dev1+ge1510fc-py3.8.egg/actinia_core/resources/common/config.py", line 334, in read
    with open(path, 'r') as configfile:
FileNotFoundError: [Errno 2] No such file or directory: '/etc/default/actinia'
```

Not sure if this is related to some install location being different than expected, or if I missed some step. I do not have any actinia folder in /etc/default.

### LINKED ISSUE #174: Install, step 7: URL not found on the server
Following instructions from here: https://actinia.mundialis.de/tutorial/installation.html
So far I believe I have gotten through steps 1-6. I have redis server running, and I started actinia-server using this command:
```
(actinia_venv) halota@basement-machine:~/actinia/workspace/tmp/actinia_core$ actinia-server --host 0.0.0.0 --port 5000
 * Serving Flask app "actinia_core.resources.common.app" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
```

Now I am following step 7 instructions, but it's not finding it.

```
halota@basement-machine:~$ export PORT=5000
halota@basement-machine:~$ export IP=http://localhost
halota@basement-machine:~$ export AUTH='-u superadmin:abcdefgh'
halota@basement-machine:~$ curl -u superadmin:abcdefgh -X GET http://localhost:5000/locations
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>404 Not Found</title>
<h1>Not Found</h1>
<p>The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.</p>
```

### LINKED ISSUE #211: Check install steps
Follow the installation steps from https://actinia.mundialis.de/tutorial/installation.html and document deprecated information in this ticket or directly in the documentation.
Attention: the documentation changed from rst files to md: https://github.com/mundialis/actinia_core/blob/main/docs/docs/installation.md

### PR DESCRIPTION
**Pull request by neteler**
This PR addresses several issues:

- update docs to use standard packages wherever possible
- clarify wording
- persistent use of Python 3 (note that Python 2 is EOL for while)
- add install and start of redis server
- add missing actinia-API (https://github.com/mundialis/actinia-api)

Fixes #170
Fixes #171
Fixes #172
Fixes #174
Fixes #211

TODO:
- address `cd /tmp` issue mentioned in #169

### CODE REVIEW

**Comment by neteler** [comment]
(seems I overlooked PR #274 - best might be to merge both somehow)

File: docs/docs/installation.md [hunk 1]
```diff
@@ -5,41 +5,40 @@ A local installation requires a full GRASS GIS installation. Make sure
 all requirements are fulfilled to compile GRASS GIS from the git
 repository.
 
-First install Proj4 and GRASS GIS by downloading the latest git version
-and compile it.
+These installation instructions a based on a Ubuntu Linux system (other
```
Comment (marcjansen) [inline docs/docs/installation.md:8]:
```suggestion
These installation instructions are based on a Ubuntu Linux system (other
```

File: docs/docs/installation.md [hunk 2]
```diff
@@ -5,41 +5,40 @@ A local installation requires a full GRASS GIS installation. Make sure
...
- wget http://download.osgeo.org/proj/proj-datumgrid-1.8.zip
+```bash
+ sudo apt-get update
+ sudo apt-get install proj-bin proj-data
+```
 
- unzip proj-datumgrid-1.8.zip
 
- cd ..
+2.  Set Python 3 environemtn
```
Comment (marcjansen) [inline docs/docs/installation.md:23]:
```suggestion
2.  Set Python 3 environment
```

File: docs/docs/installation.md [hunk 3]
```diff
@@ -117,55 +113,73 @@ and compile it.
...
+ actinia-server --host 0.0.0.0 --port 5000
 ```
 
-7.  Check the service with curl
+
+8.  Check the service with curl
 
 ```bash
-ACTINIA_VERSION="v2"
+ ACTINIA_VERSION="v2"
```
Comment (marcjansen) [inline docs/docs/installation.md:172]:
```suggestion
ACTINIA_VERSION="v2"
```

**Review by marcjansen — COMMENTED** [review]
Would we recommend using `venv`? If so, I'd prefer to have the `#` be removed, and would have this be consistently mentioned in the examples.

File: docs/docs/installation.md [hunk 4]
```diff
@@ -5,41 +5,40 @@ A local installation requires a full GRASS GIS installation. Make sure
...
- make -j4
- sudo make install
+```bash
+ # recommended to give Python3 precedence over Python2 (which is end-of-life since 2019)
+ sudo update-alternatives --install /usr/bin/python python /usr/bin/python3 1
 ```
 
-2.  Install GRASS GIS and additional modules:
+
+3.  Conmpile and install GRASS GIS and additional modules:
```
Comment (annarieger) [inline docs/docs/installation.md:31]:
```suggestion
3.  Compile and install GRASS GIS and additional modules:
```

**Comment by neteler** [comment]
> Would we recommend using `venv`? If so, I'd prefer to have the `#` be removed, and would have this be consistently mentioned in the examples.

I would entirely drop `venv` as no longer needed in Python 3 systems (I would just drop all references to Python 2, it is end-of-life for a while, see https://www.python.org/doc/sunset-python-2/)

File: docs/docs/installation.md (hunk 4 above)
Comment (neteler) [reply docs/docs/installation.md]: Thanks, @annarieger !

**Comment by marcjansen** [comment]
> > Would we recommend using `venv`? If so, I'd prefer to have the `#` be removed, and would have this be consistently mentioned in the examples.
> 
> I would entirely drop `venv` as no longer needed in Python 3 systems (I would just drop all references to Python 2, it is end-of-life for a while, see https://www.python.org/doc/sunset-python-2/)

Fine to drop `venv` completely, also from comments in this file.

I still think [`venv`](https://docs.python.org/3/library/venv.html#module-venv) is used by some python3 users, but it is totally fine for me to not mention it here in this installation documentation

**Comment by neteler** [comment]
PR updated 
- esp. `venv` completely dropped as no longer needed with Python3 being the standard nowadays
- GRASS GIS: update to production related compilation flags (no debugging flags): should be faster in data processing
- remove trailing white space from commands 

Remaining TODOs:
- address `cd /tmp` issue mentioned in #169 - suggestions?
- cherry-pick `docs/docs/actinia.cfg` changes from #274 - @Momen-Mawad @mmacata others?

**Comment by marcjansen** [comment]
> * address `cd /tmp` issue mentioned in [Install Issues, (step 2 in directions) #169](https://github.com/mundialis/actinia_core/issues/169) - suggestions?

If I understand te linked issue crrectly, this is just about that the contents of `/tmp` are gone after a reboot. If so, this tutorial/documentation can e.g. simpy state this explicitly and suggest an alternative location in case one thinks it cannot be done in one setting.

> * cherry-pick docs/docs/actinia.cfg changes from edited installation steps for actinia without docker #274 - @Momen-Mawad @mmacata others?

If only the changes to `docs/docs/actinia.cfg` are relevant once this PR merged, it might be easier to simply merge this PR, close #274 and open a new one with these changes. My 2 ¢

File: docs/docs/installation.md [hunk 5]
```diff
@@ -5,167 +5,174 @@ A local installation requires a full GRASS GIS installation. Make sure
...
- cd t.rast.aggr_func
- make
- sudo make install
-
- cd ../../display/
- git clone https://bitbucket.org/huhabla/d_rast_multi.git d.rast.multi
- cd d.rast.multi
- make
- sudo make install
+cd /tmp
```
Comment (marcjansen) [inline docs/docs/installation.md:36]:
```suggestion
# or some other location of your choice, /tmp will not keep data between reboots
cd /tmp 
```

**Comment by mmacata** [comment]
> > * cherry-pick docs/docs/actinia.cfg changes from edited installation steps for actinia without docker [edited installation steps for actinia without docker #274](https://github.com/mundialis/actinia_core/pull/274) - @Momen-Mawad @mmacata others?
> 
> If only the changes to `docs/docs/actinia.cfg` are relevant once this PR merged, it might be easier to simply merge this PR, close #274 and open a new one with these changes. My 2 ¢

I agree and would leave the `docs/docs/actinia.cfg` file out of this PR. When actinia starts up without modifying the default values, it is fine and not necessarily needed.

BTW thanks for looking into this :)

**Review by mmacata — APPROVED** [review]

**Pull request merged**